### DISCUSSION TITLE
Project planning

### DISCUSSION THREAD

**Discussion by mpeterdev** [near/react-on-chain#84] [Announcements]
A topic of much discussion lately is how to communicate progress of BWE and what milestones/deliverables look like. Each major capability of BWE is its own Epic issue in this repository, and naturally bit of a milestone unto itself. For work planning, it does not necessarily make sense to work on related items in order. Instead, we are lining work up such that we are addressing large unknowns and items with high architecture impact.

A clear example is hooks vs `useRef`. We are currently prioritizing general hook support, but `useRef` is a specific hook which will take dedicated work. It might feel natural to proceed immediately from general hooks support to any other hook-related tasks, but instead it makes sense to circle back to `useRef` after higher impact items like arbitrary dependencies (npm imports) since arbitrary dependency support raises significantly more variables in the overall implementation of BWE.

We end up with our roadmap looking like a series of items based on impact and otherwise not presenting clear bodies of work for milestones. Despite this, I think it is the most effective way to build BWE and the question is how to build a milestone framework around it.

A few approaches jump to mind, but I would be eager to hear other suggestions as well.
1. Group epics, potentially somewhat arbitrarily, into milestones for the purposes of demo and communication. e.g. every once in a while we have a check in of "Hi all, here are 5 capabilities of BWE which were implemented recently". We could also use this as a time to step back and perform a performance analysis, review the roadmap, etc
2. Choose a time period and share a regular update. "Here is what BWE has been focused on for the last month(with an demo component)"
3. Identify+label exciting epics and base communication around those. I'll use the previous examples again: Broadcasting "BWE now supports most hooks!" is much more enticing than "BWE now supports useRef!"

Regardless of how we choose to communicate progress, generating demo components for anyone to view current functionality of BWE by visiting our Vercel deployment is something we plan to stay on top of.

**Comment by charleslavon** [comment]
I agree with this sequencing of work `we are lining work up such that we are addressing large unknowns and items with high architecture impact.`

Regarding how to organize milestones, I value the sharing of updates/demos at a regular cadence.  Is this something that you have in mind for all three approaches or only #2?   

@thisisjoshford how would you feel about going roughly a month in between BWE updates?

**Reply by mpeterdev** [reply]
> Regarding how to organize milestones, I value the sharing of updates/demos at a regular cadence. Is this something that you have in mind for all three approaches or only #2?

All three, but determining whether it's a time-bound cadence or varies with work completion. I am leaning away from 2 since it seems more valuable to grab attention when we have something completed to share as opposed to posting regularly about things in progress.

We also don't need to overengineer this, I am mainly looking for to set some rough expectations for communication. We could default to `#3` but then post an in-progress update if we are working on something particularly long lived and won't have a completion milestone to share for longer than we want to wait to check in
